src/hugr/rewrite/outline_cfg.rs

```diff
@@ -0,0 +1,273 @@
+
+        // 2. New CFG node will be contained in new single-successor BB
+        let mut new_block = existing_cfg
+            .block_builder(inputs.clone(), vec![type_row![]], outputs.clone())
+            .unwrap();
+
+        // 3. new_block contains input node, sub-cfg, exit node all connected
+        let wires_in = inputs.iter().cloned().zip(new_block.input_wires());
+        let cfg = new_block.cfg_builder(wires_in, outputs).unwrap();
+        let cfg_node = cfg.container_node();
+        let inner_exit = cfg.exit_block().node();
+        let cfg_outputs = cfg.finish_sub_container().unwrap().outputs();
+        let predicate = new_block
+            .add_constant(ConstValue::simple_predicate(0, 1))
+            .unwrap();
+        let pred_wire = new_block.load_const(&predicate).unwrap();
+        let new_block = new_block
+            .finish_with_outputs(pred_wire, cfg_outputs)
+            .unwrap();
+
+        // 4. Entry edges. Change any edges into entry_block from outside, to target new_block
+        let h = existing_cfg.hugr_mut();
+
+        let preds: Vec<_> = h
+            .linked_ports(entry, h.node_inputs(entry).exactly_one().unwrap())
+            .collect();
+        for (pred, br) in preds {
+            if !self.0.contains(&pred) {
+                h.disconnect(pred, br).unwrap();
+                h.connect(pred, br.index(), new_block.node(), 0).unwrap();
+            }
+        }
+
+        // 5. Children of new CFG.
+        // Entry node must be first
+        h.hierarchy.detach(entry.index);
+        h.hierarchy
+            .insert_before(entry.index, inner_exit.index)
+            .unwrap();
+        // And remaining nodes
+        for n in self.0 {
+            // Do not move the entry node, as we have already
+            if n != entry {
+                h.hierarchy.detach(n.index);
+                h.hierarchy.push_child(n.index, cfg_node.index).unwrap();
+            }
+        }
+
+        // 6. Exit edges.
+        // Retarget edge from exit_node (that used to target outside) to inner_exit
+        let exit_port = h
+            .node_outputs(exit)
+            .filter(|p| {
+                let (t, p2) = h.linked_ports(exit, *p).exactly_one().ok().unwrap();
+                assert!(p2.index() == 0);
+                t == outside
+            })
+            .exactly_one()
+            .unwrap();
+        h.disconnect(exit, exit_port).unwrap();
+        h.connect(exit, exit_port.index(), inner_exit, 0).unwrap();
+        // And connect new_block to outside instead
+        h.connect(new_block.node(), 0, outside, 0).unwrap();
+
+        Ok(())
+    }
+}
+
+/// Errors that can occur in expressing an OutlineCfg rewrite.
+#[derive(Debug, Error)]
+pub enum OutlineCfgError {
+    /// The set of blocks were not siblings
+    #[error("The nodes did not all have the same parent")]
+    NotSiblings,
+    /// The parent node was not a CFG node
+    #[error("The parent node {0:?} was not a CFG but an {1:?}")]
```

Comment:
```suggestion
    #[error("The parent node {0:?} was not a CFG but an {1:?}")]
```
"a" or "an"? I don't propose going to great lengths to solve this, unless you can think of a phrasing that avoids the question.

Reply:
But let's at least be consistent ;-). Gone for "a".